src/quest_manager.py

```
@@ -182,5 +190,135 @@ def generate(self) -> Optional[Quest]:
+        Args:
+            event_id: ID of the event to end
+            end_notes: Optional notes to append about how the event ended
+
+        Returns:
+            The updated TaskEvents object or None if update failed
+        """
+        try:
+            event.end_time = datetime.now()
+
+            # If there are end notes, append them to the existing notes file
+            if end_note != None:
```

Ruff (0.8.2)

Undefined name `end_note`

(F821)